File: java/api/src/main/java/eu/ill/cameo/api/base/CancelEvent.java
/*
 * CAMEO
 *
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under BSD 3-Clause and GPL-v3 as described in license files.
 * You may not use this work except in compliance with the Licences.
 *
 */

package eu.ill.cameo.api.base;

import org.json.simple.JSONObject;

/**
 * Class defining a cancel event.
 */
public class CancelEvent extends Event {
	
	/**
	 * Constructor.
	 * @param id The application id.
	 * @param name The application name.
	 */
	public CancelEvent(int id, String name) {
		super(id, name);
	}
		
	@Override
	public boolean equals(Object obj) {
		if (this == obj) {
			return true;
		}
		if (obj == null) {
			return false;
		}
		if (getClass() != obj.getClass()) {
			return false;
		}
		CancelEvent other = (CancelEvent) obj;
		
		if (id != other.id) {
			return false;
		}
		if (name == null) {
			if (other.name != null) {
				return false;
			}
		}
		else if (!name.equals(other.name)) {
			return false;
		}
		return true;
	}
	
	@Override
	public String toString() {
		JSONObject result = new JSONObject();
		
		result.put("type", "cancel");
		result.put("id", id);
		result.put("name", name);
		
		return result.toJSONString();
	}
	
}

File: cpp/api/src/base/CancelEvent.cpp
/*
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under the EUPL, Version 1.1 only (the "License");
 * You may not use this work except in compliance with the Licence.
 * You may obtain a copy of the Licence at:
 *
 * http://joinup.ec.europa.eu/software/page/eupl
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the Licence is distributed on an "AS IS" basis,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the Licence for the specific language governing permissions and
 * limitations under the Licence.
 */

#include "CancelEvent.h"
#include "JSON.h"
#include <iostream>

namespace cameo {

CancelEvent::CancelEvent(int id, const std::string& name) :
	Event(id, name) {
}

CancelEvent::CancelEvent(const CancelEvent& event) :
	Event(event) {
}

CancelEvent* CancelEvent::clone() {
	return new CancelEvent(*this);
}

std::string CancelEvent::toString() const {
	json::StringObject jsonObject;

	jsonObject.pushKey("type");
	jsonObject.pushValue("cancel");

	jsonObject.pushKey("id");
	jsonObject.pushValue(m_id);

	jsonObject.pushKey("name");
	jsonObject.pushValue(m_name);

	return jsonObject.dump();
}

std::ostream& operator<<(std::ostream& os, const cameo::CancelEvent& event) {
	os << event.toString();

	return os;
}

}
